build/configs/artik053/Make.defs

```diff
@@ -126,19 +151,32 @@ endif
 endif
 
 endif
-
 ARCHDEFINES =
 ARCHPICFLAGS = -fpic -msingle-pic-base -mpic-register=r10
 
 CFLAGS = $(ARCHCFLAGS) $(ARCHWARNINGS) $(ARCHOPTIMIZATION) $(ARCHCPUFLAGS) $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRADEFINES) -pipe -ffunction-sections -fdata-sections
 CPICFLAGS = $(ARCHPICFLAGS) $(CFLAGS)
-CXXFLAGS  = $(ARCHCXXFLAGS) $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) $(ARCHCPUFLAGS) $(ARCHXXINCLUDES)
-CXXFLAGS += $(ARCHDEFINES) $(EXTRADEFINES) -pipe -std=c++11 -DCLOCK_MONOTONIC -D__TINYARA__
-CXXFLAGS += -fexceptions -fcheck-new -frtti
-CXXFLAGS += -pedantic -D_DEBUG -D_LIBCPP_BUILD_STATIC -ffunction-sections -fdata-sections -D__GLIBCXX__
+CXXFLAGS = $(ARCHCXXFLAGS) $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) $(ARCHCPUFLAGS) $(ARCHXXINCLUDES) $(ARCHDEFINES) $(EXTRADEFINES) -pipe
 ifeq ($(CONFIG_LIBCXX),y)
 CXXFLAGS += -DCONFIG_WCHAR_BUILTIN
 endif
+
+ifneq ($(filter y,$(CONFIG_HAVE_CXX) $(CONFIC_GRPC)),)
+CXXFLAGS += -std=c++11 -DCLOCK_MONOTONIC -D__TINYARA__
+CXXFLAGS += -fcheck-new -D_DEBUG -D_LIBCPP_BUILD_STATIC -ffunction-sections -fdata-sections
+endif
+
+ifeq ($(CONFIG_HAVE_CXX),y)
+CXXFLAGS += -fexceptions -frtti
+CXXFLAGS += -pedantic -D__GLIBCXX__
+
+endif
+
+ifeq ($(CONFIC_GRPC),y)
+CXXFLAGS += -fno-exceptions  -fno-rtti
+CXXFLAGS += -D_LIBCPP_NO_EXCEPTIONS
+endif
```

Comment:
If both are enabled, this will add opposite options. 
This is related with previous comment.
```
This looks strange.
When CONFIG_HAVE_CXX is enabled, will not CONFIG_GRPC be enabled? I think they are independent.
```